qkit/measure/timedomain/sequence_library.py

```diff
@@ -0,0 +1,122 @@
+"""
+sequency_library.py
+A. Stehli (04/2018)
+
+This library provides the measurement sequences for standard qubit experiments.
+Currently this includes:
+    rabi experiment
+    T1 measurments
+    Ramsey experiments
+    Spin/Hahn echo
+"""
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+import pulse_sequence as ps
+
+
+def compensate():
+    # This should be moved to sequence class.
+    # Sequence class could have attributes .compensate, .decaytime.
+    # If compensate: sequence.__call__ accounts for the decay due to decaytime.
+    return
+
+def rabi(sample, pulse_shape = ps.ShapeLib.rect, amplitude = 1, frequency_shift = 0, sample_rate = None):
+    """
+        Generate sequence object for a rabi experiment (varying drive pulse lengths).
+
+        Input:
+            sample          - sample object.
+            pulse_shape     - Shape of the pulses (i.e. square, gauss, ...)
+            amplitude       - relative ampitude of the sequence.
+            frequency_shift - frequency shift of the pulses.
+
+        Output:
+            Sequence object for rabi experiment.
+    """
+    rabi_tone = ps.Pulse(lambda t: t, shape = pulse_shape, name = "rabi-tone", amplitude = amplitude, frequency_shift = frequency_shift)
+    sequence = ps.PulseSequence(sample, sample.clock)
```

Comment:
passing `sample.clock` is superfluous

The same applies for all experiments below

Reply:
Fixed